orion/handlers/http/defaults.go

```diff
@@ -37,25 +38,22 @@ func DefaultEncoder(req *http.Request, r interface{}) error {
 	if req.Method == http.MethodGet && len(data) == 0 {
 		return nil
 	}
-	return deserialize(req.Context(), data, r)
+	return deserialize(req.Context(), data, r, JSONPB)
 }
 
-func deserialize(ctx context.Context, data []byte, r interface{}) error {
+func deserialize(ctx context.Context, data []byte, r interface{}, JSONPB bool) error {
 	serType := ContentTypeFromHeaders(ctx)
-	switch serType {
-	case modifiers.ProtoBuf:
+	if serType == modifiers.ProtoBuf {
 		if protoReq, ok := r.(proto.Message); ok {
 			return proto.Unmarshal(data, protoReq)
 		}
-		fallthrough
-	case modifiers.JSONPB:
+	}
+	if serType == modifiers.JSONPB || JSONPB {
```

Comment:
Let's make the logic here more explicit, in case we choose to add new serialization types in the future.

Instead of `content-type == "application/jsonpb" || defaultJSONPB`, 
change to 
`content-type == "application/jsonpb" || (content-type == "application/json" && defaultJSONPB)`